test/runtests.jl

```diff
@@ -40,7 +40,11 @@ non_diffeable_arg_functions = [(:Base, :rem2pi, 2), (:Base, :ldexp, 2), (:Base,
                     # We're happy with types with the correct promotion behavior, e.g.
                     # it's fine to return `1` as a derivative despite input being `Float64`.
                     @test promote_type(typeof($deriv), $T) === $T
-                    @test $deriv ≈ finitediff($M.$f, goo) rtol=1e-2 atol=1e-3
+                    if $(f in (:log1pmx, :logmxp1)) && $T == Float32
+                        @test_throws MethodError finitediff($M.$f, goo)
+                    else
+                        @test $deriv ≈ finitediff($M.$f, goo) rtol=1e-2 atol=1e-3     
```

Comment:
Seems good. Can you add `logmxp1` to the `rand($T) + $T(0.5)` branch above? That should make finite differencing even less likely to error.

Reply:
Will do!